python/ray/libraylet_library_python.pyx

```diff
@@ -0,0 +1,723 @@
+        return return_id_list
+
+    def required_resources(self):
+        """Return the resource vector of the task."""
+        cdef:
+            unordered_map[c_string, double] resource_map = self.task_spec.get().GetRequiredResources().GetResourceMap()
+            c_string resource_name
+            double resource_value
+            unordered_map[c_string, double].iterator iterator = resource_map.begin()
+
+        required_resources = {}
+        while iterator != resource_map.end():
+            resource_name = dereference(iterator).first
+            py_resource_name = str(resource_name)  # bytes for Py2, unicode for Py3
+            resource_value = dereference(iterator).second
+            required_resources[py_resource_name] = resource_value
+            postincrement(iterator)
+        return required_resources
+
+    def actor_creation_id(self):
+        """Return the actor creation ID for the task."""
+        return PyActorID.from_native(self.task_spec.get().ActorCreationId())
+
+    def actor_creation_dummy_object_id(self):
+        """Return the actor creation dummy object ID for the task."""
+        return PyObjectID.from_native(self.task_spec.get().ActorCreationDummyObjectId())
+
+    def actor_id(self):
+        """Return the actor ID for this task."""
+        return PyActorID.from_native(self.task_spec.get().ActorId())
+
+    def actor_counter(self):
+        """Return the actor counter for this task."""
+        return self.task_spec.get().ActorCounter()
+
+    def _serialized_raylet_task(self):
+        """This is a hack used to create a serialized flatbuffer object for the
+        driver task. We're doing this because creating the flatbuffer object in
+        Python didn't seem to work.
+        """
+        cdef:
+            # Classes without a nullary constructor cannot be allocated in stack in Cython
+            RayletTaskExecutionSpecification* execution_spec = new RayletTaskExecutionSpecification(move(self.execution_dependencies.get()[0]))
+            RayletTask* task = new RayletTask(execution_spec[0], self.task_spec.get()[0])
+            FlatBufferBuilder fbb
+            Offset[_Task] task_flatbuffer = task[0].ToFlatbuffer(fbb)
+        fbb.Finish(task_flatbuffer)
+        out_str = fbb.GetBufferPointer()[:fbb.GetSize()]
+        del task
+        del execution_spec
+        return out_str
+
+
+cdef class RayletClient:
+    cdef unique_ptr[LibRayletClient] client
+    def __cinit__(self, raylet_socket,
+                  PyClientID client_id,
+                  c_bool is_worker,
+                  PyJobID driver_id):
+        cdef:
+            c_string c_raylet_socket = to_c_string(raylet_socket)
+        # We have known that we are using Python, so just skip the language parameter.
+        self.client.reset(new LibRayletClient(c_raylet_socket, client_id.data,
+                          is_worker, driver_id.data, LANGUAGE_PYTHON))
+
+    def disconnect(self):
+        cdef:
```

Comment:
having this on the same line without the colon makes sense if there is only one declaration (same above)